src/main/java/randoop/util/Randomness.java

```diff
@@ -36,21 +38,54 @@ public static int nextRandomInt(int i) {
     if (Log.isLoggingOn()) {
       Log.logLine("randoop.util.Randomness: " + totalCallsToRandom + " calls so far.");
     }
-    return random.nextInt(i);
+    int value = random.nextInt(i);
+    if (GenInputsAbstract.selection_log != null) {
+      StackTraceElement[] trace = Thread.currentThread().getStackTrace();
```

Comment:
This code is repeated almost identically multiple times in the file.  Could you please abstract it out into a method?  That will also make it easier to call from elsewhere.